x-pack/legacy/plugins/ml/server/routes/data_frame_analytics.js

```diff
@@ -4,171 +4,302 @@
+    licensePreRoutingFactory(xpackMainPlugin, dfAnalyticsExplainHandler)
+  );
 
-  route({
-    method: 'POST',
-    path: '/api/ml/data_frame/analytics/{analyticsId}/_start',
-    handler(request) {
-      const callWithRequest = callWithRequestFactory(elasticsearchPlugin, request);
-      const options = {
-        analyticsId: request.params.analyticsId,
-      };
+  async function dfAnalyticsDeleteJobHandler(context, request, response) {
+    try {
+      const { analyticsId } = request.params;
+      const results = await context.ml.mlClient.callAsCurrentUser('ml.deleteDataFrameAnalytics', {
+        analyticsId,
+      });
+      return response.ok({
+        body: { ...results },
+      });
+    } catch (e) {
+      // Case: default
+      return response.internalError({ body: e.message });
+    }
+  }
 
-      return callWithRequest('ml.startDataFrameAnalytics', options).catch(resp => wrapError(resp));
+  router.delete(
+    {
+      path: '/api/ml/data_frame/analytics/{analyticsId}',
+      validate: {
+        params: schema.object({
+          analyticsId: schema.string(),
+        }),
+      },
     },
-    config: {
-      ...commonRouteConfig,
+    licensePreRoutingFactory(xpackMainPlugin, dfAnalyticsDeleteJobHandler)
+  );
+
+  async function dfAnalyticsStartJobHandler(context, request, response) {
+    try {
+      const { analyticsId } = request.params;
+      const results = await context.ml.mlClient.callAsCurrentUser('ml.startDataFrameAnalytics', {
+        analyticsId,
+      });
+      return response.ok({
+        body: { ...results },
+      });
+    } catch (e) {
+      // Case: default
+      return response.internalError({ body: e.message });
+    }
+  }
+
+  router.post(
+    {
+      path: '/api/ml/data_frame/analytics/{analyticsId}/_start',
+      validate: {
+        params: schema.object({
+          analyticsId: schema.string(),
+        }),
+      },
     },
-  });
+    licensePreRoutingFactory(xpackMainPlugin, dfAnalyticsStartJobHandler)
+  );
 
-  route({
-    method: 'POST',
-    path: '/api/ml/data_frame/analytics/{analyticsId}/_stop',
-    handler(request) {
-      const callWithRequest = callWithRequestFactory(elasticsearchPlugin, request);
+  async function dfAnalyticsStopJobHandler(context, request, response) {
+    try {
       const options = {
         analyticsId: request.params.analyticsId,
       };
 
-      if (request.query.force !== undefined) {
-        options.force = request.query.force;
+      if (request.url?.query?.force !== undefined) {
+        options.force = request.url.query.force;
       }
 
-      return callWithRequest('ml.stopDataFrameAnalytics', options).catch(resp => wrapError(resp));
-    },
-    config: {
-      ...commonRouteConfig,
+      const results = await context.ml.mlClient.callAsCurrentUser(
+        'ml.stopDataFrameAnalytics',
+        options
+      );
+      return response.ok({
+        body: { ...results },
+      });
+    } catch (e) {
+      // Case: default
```

Comment:
NIT: comment seems strange?

Reply:
Removed in 1be4d620155fc0186cd7fedcaebb6c7ba5a51c9c